http.c

```diff
@@ -3719,6 +3768,13 @@ evhttp_set_allowed_methods(struct evhttp* http, ev_uint16_t methods)
 	http->allowed_methods = methods;
 }
 
+void
+evhttp_set_extended_methods(struct evhttp* http,
+    const struct evhttp_extended_methods *ext_methods)
+{
+	http->ext_methods = ext_methods;
```

Comment:
I think I see what you mean : people may pass a stack allocated array, or change its content.
The copy will add malloc/free, I need to think about it